### PR TITLE
removed kitchen_cleanup.yml for E2E project

### PR DESCRIPTION
**Pull request by amitslavin**
### What does this PR do?
Removing the kitchen_cleanup in order to use the images for E2E testing

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by usamasaqib — COMMENTED** [review]
Can you give some more context here? How does removing this job help with E2E testing? Will you be replacing it with something else?

**Review by KSerrania — CHANGES_REQUESTED** [review]

**Comment by amitslavin** [comment]
> Can you give some more context here? How does removing this job help with E2E testing? Will you be replacing it with something else?

@usamasaqib 
As I understood the purpose is to use the packages for testing - to be able to manually trigger a specific package. [link](https://datadoghq.atlassian.net/browse/AETT-130) to the task. 
Also I understood that there is a job which is deleting the packages every 24 hours which will still remove old packages.

**Comment by pducolin** [comment]
> As I understood the purpose is to use the packages for testing - to be able to manually trigger a specific package. [link](https://datadoghq.atlassian.net/browse/AETT-130) to the task.

Indeed, current flow is (1) build -> (2) kitchen tests -> (3) build_cleanup

We need to remove step (3) to use artifacts from step (1) in manually triggered automated e2e tests

**Review by pducolin — APPROVED** [review]
Thank you !

**Review by pducolin — APPROVED** [review]
Thank you !

**Comment by amitslavin** [comment]
@pducolin thanks,

I was chatting with @usamasaqib and he was interested regarding the job which should remove kitchen tests after 24 hours, he suspects that it may clean only jobs which are running every 24 hours

File: .gitlab/kitchen_cleanup.yml [hunk 1]
```diff
@@ -1,41 +0,0 @@
...
-    !reference [.on_kitchen_tests_a6_always]
-  variables:
-    DD_PIPELINE_ID: $CI_PIPELINE_ID-a6
-
-kitchen_cleanup_azure-a7:
-  extends: .kitchen_cleanup_azure_common
-  rules:
-    !reference [.on_default_kitchen_tests_a7_always]
-  variables:
-    DD_PIPELINE_ID: $CI_PIPELINE_ID-a7
```
[the remark this answers is not in the document]
Comment (amitslavin) [reply .gitlab/kitchen_cleanup.yml]: Done, thank you for your clarification!

**Comment by pducolin** [comment]
@amitslavin , @usamasaqib here is the daily cleanup job: https://github.com/DataDog/datadog-agent/blob/main/.gitlab/maintenance_jobs/kitchen.yml#L2-L16

It removes everything in s3

**Comment by amitslavin** [comment]
Thank you for sharing @pducolin

**Review by KSerrania — APPROVED** [review]
LGTM

**Review by usamasaqib — APPROVED** [review]

**Pull request merged**